packages/core/utils/src/dal/mikro-orm/base-entity.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+} from "@mikro-orm/core"
+import { generateEntityId } from "../../common"
+
+@Entity({ abstract: true })
+export class BaseEntity {
+  [OptionalProps]?: BaseEntity["id"] | BaseEntity["__prefix_id__"]
+
+  private __prefix_id__?: string
+
+  constructor({ prefix_id }: { prefix_id?: string } = {}) {
+    this.__prefix_id__ = prefix_id
+  }
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @OnInit()
+  @BeforeCreate()
+  onInit() {
+    this.id ??= this.generateEntityId(this.__prefix_id__)
+  }
+
+  private generateEntityId(prefixId?: string): string {
+    let ensuredPrefixId: string = prefixId as string
+
+    if (ensuredPrefixId) {
+      return generateEntityId(undefined, ensuredPrefixId)
+    }
+
+    ensuredPrefixId = Object.getPrototypeOf(this).constructor.name
+
+    /*
+     * Split the class name (camel case) into words and exclude model and entity from the words
+     */
+    const words = ensuredPrefixId
+      .split(/(?=[A-Z])/)
+      .filter((word) => !["entity", "model"].includes(word.toLowerCase()))
+    const wordsLength = words.length
+
+    /*
+     * if the class name (camel case) contains one word, the prefix id is the first three letters of the word
+     * if the class name (camel case) contains two words, the prefix id is the first two letters of the first word plus the first letter of the second one
+     * if the class name (camel case) contains more than two words, the prefix id is the first letter of each word
```

Comment:
I am not sure how readable this case would be, but I guess is fine since you can override it anyway.

Reply:
yes thats the idea, it is mainly for a user that does not want to provide a prefix